[2024-07-18T15:04:30.739+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: google_analytics_data_etl.task_transform_to_flat_by_operation manual__2024-07-18T15:04:26.291451+00:00 [queued]>
[2024-07-18T15:04:30.749+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: google_analytics_data_etl.task_transform_to_flat_by_operation manual__2024-07-18T15:04:26.291451+00:00 [queued]>
[2024-07-18T15:04:30.750+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 2
[2024-07-18T15:04:30.765+0000] {taskinstance.py:2192} INFO - Executing <Task(PythonOperator): task_transform_to_flat_by_operation> on 2024-07-18 15:04:26.291451+00:00
[2024-07-18T15:04:30.775+0000] {standard_task_runner.py:60} INFO - Started process 841 to run task
[2024-07-18T15:04:30.781+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'google_analytics_data_etl', 'task_transform_to_flat_by_operation', 'manual__2024-07-18T15:04:26.291451+00:00', '--job-id', '148', '--raw', '--subdir', 'DAGS_FOLDER/google_analytics_data_etl/google_analytics_data_etl.py', '--cfg-path', '/tmp/tmprelei6g4']
[2024-07-18T15:04:30.783+0000] {standard_task_runner.py:88} INFO - Job 148: Subtask task_transform_to_flat_by_operation
[2024-07-18T15:04:30.836+0000] {task_command.py:423} INFO - Running <TaskInstance: google_analytics_data_etl.task_transform_to_flat_by_operation manual__2024-07-18T15:04:26.291451+00:00 [running]> on host 3d44d7374020
[2024-07-18T15:04:30.983+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='google_analytics_data_etl' AIRFLOW_CTX_TASK_ID='task_transform_to_flat_by_operation' AIRFLOW_CTX_EXECUTION_DATE='2024-07-18T15:04:26.291451+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-07-18T15:04:26.291451+00:00'
[2024-07-18T15:04:31.120+0000] {xcom.py:664} ERROR - keys must be str, int, float, bool or None, not tuple. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-07-18T15:04:31.125+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/google_analytics_data_etl/logics/transform.py", line 60, in transform_to_flat_by_operation
    ti.xcom_push(key='transform_flat_group_event_name', value=data_to_dict)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2981, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 662, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
TypeError: keys must be str, int, float, bool or None, not tuple
[2024-07-18T15:04:31.162+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=google_analytics_data_etl, task_id=task_transform_to_flat_by_operation, execution_date=20240718T150426, start_date=20240718T150430, end_date=20240718T150431
[2024-07-18T15:04:31.176+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 148 for task task_transform_to_flat_by_operation (keys must be str, int, float, bool or None, not tuple; 841)
[2024-07-18T15:04:31.216+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-07-18T15:04:31.246+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
